e2e/benchmarks/services_test.go

```diff
@@ -0,0 +1,276 @@
+// +build e2e
+
+// Runs benchmarks using the Null provider and working with Consul setup with
+// multiple Services or Service Instances.
+
+package benchmarks
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/command"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const configFile = "config.hcl"
+const tempDirPrefix = "tmp_"
+
+// Benchmark N Services Instance (all the same service)
+func Benchmark1Instance(b *testing.B) {
+	benchmarkInstances(b, 1)
+}
+func Benchmark100Instances(b *testing.B) {
+	benchmarkInstances(b, 100)
+}
+func Benchmark1000Instances(b *testing.B) {
+	benchmarkInstances(b, 1000)
+}
+func benchmarkInstances(b *testing.B, N int) {
+	srv := testutils.NewTestConsulServerHTTPS(b, "../../testutils")
+	defer srv.Stop()
+	path, cleanup := setupServiceInstances(b, srv, N)
+	defer cleanup() // comment out if you want to keep the config files
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		require.NoError(b, runSyncOnce(path))
+	}
+}
+
+// Benchmark N Services
+func Benchmark1Service(b *testing.B) {
+	benchmarkServices(b, 1)
+}
+func Benchmark100Service(b *testing.B) {
+	benchmarkServices(b, 1)
+}
+func Benchmark1000Service(b *testing.B) {
+	benchmarkServices(b, 1)
+}
+func benchmarkServices(b *testing.B, N int) {
```

Comment:
These services benchmarks do not seem to be testing that CTS will monitor N services, the task itself is only monitoring one service "stress".

```
task {
  name = "stress-test"
  source = "../../tmp_stress/module"
  providers = ["local"]
  services = ["stress"]
}
```

The task would need to resemble the below to monitor all N services

```
task {
  services = ["service001", "service002", ...]
}

Reply:
Your right. I worked on the instances version at first and was still thinking along those lines when I added the services. It needs that added.